app/models/note.rb

```diff
@@ -38,18 +38,31 @@ class Note < ActiveRecord::Base
 
   class << self
     def in_or_layer_below(group)
-      joins('LEFT JOIN roles ' \
-            "ON roles.person_id = notes.subject_id AND notes.subject_type = '#{Person.sti_name}'").
-        joins("INNER JOIN #{Group.quoted_table_name} " \
-              "ON (#{Group.quoted_table_name}.id = notes.subject_id "\
-                  "AND notes.subject_type = '#{Group.sti_name}') " \
-              "OR (#{Group.quoted_table_name}.id = roles.group_id)").
+      joins(people_join).
+        joins(groups_join).
```

Comment:
Hier habe ich eine schönere Lösung gefunden, ich hatte Typos und Überlegungsfehler in meinem Fix drin. Der Test ist zu recht fehlgeschlagen.